File: frontend/portfolio-frontend/src/app/core/models/skill.ts
export interface Skill {
  id?: number;
  name: string;
  category: string;
  displayOrder: number;
}

export interface SkillsGrouped {
  LANGUAGE?: Skill[];
  FRONTEND?: Skill[];
  BACKEND?: Skill[];
  DATABASE?: Skill[];
  DEVOPS?: Skill[];
  TOOLS?: Skill[];
  [category: string]: Skill[] | undefined;
}

File: backend/src/main/java/com/shubham/portfolio/model/Skill.java
package com.shubham.portfolio.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Entity
@Table(name = "skills")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Skill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(nullable = false)
    private String name;

    @NotNull
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Category category;

    @Column(name = "display_order")
    private Integer displayOrder;

    public enum Category {
        LANGUAGE,
        FRONTEND,
        BACKEND,
        DATABASE,
        DEVOPS,
        TOOLS
    }
}
